helix-core/src/main/java/org/apache/helix/tools/commandtools/ZkGrep.java

```diff
@@ -463,9 +463,8 @@ static File gunzip(File zipFile) {
       return outputFile;
     } catch (IOException e) {
       LOG.error("fail to gunzip file: " + zipFile, e);
+      throw new Exception("fail to gunzip file" + zipFile);
```

Comment:
The Exception message should contains the original  IOException ?

Reply:
yes.

Tooling is fine here as NPE is caught outside, and proper error message are printed out.

BTW, wrapping IOException using generic Exception will erase the proper semantics that IOException carries, which is not a good practice.

Reply:
why not we check the null in ZkGrep#processCommandLineArgs:
if (lastZkSnapshot == null){
  LOG.error("fail to gunzip file" + zkDataDirs[1]);
   System.exit(1);
}

@brettKK

Reply:
@lujiefsi might not be a good idea to only check lastZkSnapshot, because gunzip() is used in multiple places and you need to check all of them.